www/docs/guides/migrate_v7_v8.md

```diff
@@ -0,0 +1,536 @@
+});
+```
+
+### Transaction Tips
+
+Starknet 0.14 introduces a **tip mechanism** for transaction prioritization in the mempool:
+
+- Transactions can include tips to prioritize execution
+- Higher tips increase the likelihood of faster inclusion
+- Tips are separate from transaction fees and go to the sequencer
+
+### Block State Changes
+
+**Important block handling changes:**
+
+- ❌ **Pending blocks have been removed** from the protocol
+- ✅ **New decentralized pre-confirmation state** replaces pending blocks
+- Block statuses are now: `PRE_CONFIRMED` → `ACCEPTED_ON_L2` → `ACCEPTED_ON_L1`
+
+### Transaction Waiting Behavior
+
+**Starknet.js v8 now waits for transactions to reach `ACCEPTED_ON_L2` status:**
+
+- `waitForTransaction()` now waits for `ACCEPTED_ON_L2` instead of pending confirmation
+
+```typescript
+// v8 behavior - waits for ACCEPTED_ON_L2
+const txReceipt = await account.waitForTransaction(txHash);
+// Transaction is now confirmed on L2
+```
+
+This affects how you handle transaction states and block confirmations in your applications.
+
+## Overview
+
+Starknet.js v8 introduces several breaking changes while adding full support for these Starknet 0.14 protocol changes and RPC 0.9. The most significant change is the move from argument-based constructors to object-based APIs for better developer experience and extensibility.
+
+## Node.js Requirements
+
+Starknet.js v8 has a soft requirement for **Node.js >= 22** due to built-in WebSocket support.
+
+- **Node.js >= 22**: WebSocket support is built-in, no additional dependencies needed
+- **Node.js < 22**: You can still use older versions but will need to provide a WebSocket implementation
+
+For older Node.js versions, install the `ws` library and configure it:
+
+```bash
+npm install ws
+npm install --save-dev @types/ws
+```
+
+```typescript
+import { WebSocketChannel, config } from 'starknet';
+import WebSocket from 'ws';
+
+// Option 1: Configure WebSocket globally using config singleton
+config.set('websocket', WebSocket);
+
+// Option 2: Pass WebSocket to WebSocketChannel constructor
+const channel = new WebSocketChannel({
+  nodeUrl: 'wss://...',
+  websocket: WebSocket,
+});
+```
+
+## RPC Version Support
+
+Starknet.js v8 introduces support for **RPC 0.9** and maintains compatibility with **RPC 0.8**. **RPC 0.7 support has been removed.**
+
+|                       |   RPC 0.7   |    RPC 0.8    | RPC 0.9 <br /> (new) |
+| --------------------: | :---------: | :-----------: | :------------------: |
+| V3 tx (Starknet fees) |     ❌      |       ✓       |          ✓           |
+|    **Support Status** | **Removed** | **Supported** |    **Supported**     |
+
+## Breaking Changes
+
+### RPC 0.7 Support Removed
+
+**Important:** Starknet.js v8 no longer supports RPC 0.7. If you were using RPC 0.7 endpoints, you must upgrade to RPC 0.8 or RPC 0.9.
+
+**v8 (use RPC 0.8 or 0.9):**
+
+```typescript
+// Option 1: Use RPC 0.8
+const provider = new RpcProvider({
```

Comment:
Would it make sense to advocate for using `RpcProvider.create(...)` in the migration guide?